src/test/java/org/springframework/data/elasticsearch/junit/jupiter/ElasticsearchRestTemplateConfiguration.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.junit.jupiter;
+
+import java.time.Duration;
+
+import org.elasticsearch.client.RestHighLevelClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.elasticsearch.client.ClientConfiguration;
+import org.springframework.data.elasticsearch.client.RestClients;
+import org.springframework.data.elasticsearch.config.AbstractElasticsearchConfiguration;
+
+/**
+ * Configuration for Spring Data Elasticsearch using
+ * {@link org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate}. The required
+ * {@link ClusterConnectionInfo} bean must be provided by the testclass.
+ *
+ * @author Peter-Josef Meisch
+ */
+@Configuration
+public class ElasticsearchRestTemplateConfiguration extends AbstractElasticsearchConfiguration {
+
+	@Autowired private ClusterConnectionInfo clusterConnectionInfo;
```

Comment:
I wouldn't make this a required bean. If it's not present in the `ApplicationContext`, can't you just create a default instance for the user?

That would also remove the requirement that the user always configure a `@Bean` method for that if the defaults are sufficient.